From 3c4907286e8884564d5ad9e2cf19a6d1cddd16b2 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 28 Feb 2017 15:08:53 +0800
Subject: [PATCH 1648/5242] MLK-14286-2 ARM: dts: imx7ulp-evk: add dts support
 for ADV7535

commit  95a25706eb9ae103da6c6fa3b063de552bd87530 from
https://source.codeaurora.org/external/imx/linux-imx.git

ADV7535 is a low-power MIPI-DSI receiver with HDMI 1.4 compliant
transmitter. And it's an I2C device attached by lpi2c5.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 0bbfb671445d1f2d5037ea7168ebcb4699760095)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 708fc49..984fcde 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -329,6 +329,12 @@
 				IMX7ULP_PAD_PTE6__PTE6		0x20043 /* WL_REG_ON */
 			>;
 		};
+
+		pinctrl_dsi_hdmi: dsi_hdmi_grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC18__PTC18	0x10103		/* DSI_HDMI_INT */
+			>;
+		};
 	};
 };
 
@@ -398,6 +404,21 @@
 	pinctrl-1 = <&pinctrl_lpi2c5>;
 	status = "okay";
 
+	adv7535: adv7535@3d {
+		compatible = "adi,adv7535";
+		reg = <0x3d>; /* PD pin is low */
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_dsi_hdmi>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <18 IRQ_TYPE_EDGE_FALLING>;
+		video-mode = <1>; /*
+			           * Only support CEA modes.
+			           * Reference mxc_edid.c
+			           */
+		bpp = <24>;
+		status = "disabled";
+	};
+
 	fxas2100x@20 {
 		compatible = "fsl,fxas2100x";
 		reg = <0x20>;
-- 
1.7.9.5

